File: input/1_list_devices.py
# How to retrieve information on the system input devices

import harfang as hg

device_typename = ['Unknown', 'Keyboard', 'Mouse', 'Pad', 'Touch', 'HMD', 'Controller']

# retrieve the list of all available devices on the system
devices = hg.GetInputSystem().GetDevices()

# print it to screen
print("The following devices are available on this system:")

for device_name in devices:
	device_type = hg.GetInputSystem().GetDevice(device_name).GetType()
	print('- %s (name: "%s")' % (device_typename[device_type], device_name))

File: input/1_list_devices.lua
-- How to retrieve information on the system input devices

hg = require("harfang")

device_typename = {'Unknown', 'Keyboard', 'Mouse', 'Pad', 'Touch', 'HMD', 'Controller'}

-- retrieve the list of all available devices on the system
devices = hg.GetInputSystem():GetDevices()

-- print it to screen
print("The following devices are available on this system:")

for i=0,(devices:size()-1) do
	local device_name = devices:at(i)
	local device_type = hg.GetInputSystem():GetDevice(device_name):GetType()
	print(string.format('- %s (name: "%s")', device_typename[device_type+1], device_name))
end